RecoTauTag/RecoTau/interface/AntiElectronIDMVA6.h

```diff
@@ -0,0 +1,166 @@
+                   Float_t TauEmFraction,
+                   Float_t TauLeadPFChargedHadrHoP,
+                   Float_t TauLeadPFChargedHadrEoP,
+                   Float_t TauVisMassIn,
+                   Float_t TaudCrackEta,
+                   Float_t TaudCrackPhi,
+                   Float_t TauHasGsf,
+                   Int_t TauSignalPFGammaCandsIn,
+                   Int_t TauSignalPFGammaCandsOut,
+                   Float_t TauGammaEtaMomIn,
+                   Float_t TauGammaEtaMomOut,
+                   Float_t TauGammaPhiMomIn,
+                   Float_t TauGammaPhiMomOut,
+                   Float_t TauGammaEnFracIn,
+                   Float_t TauGammaEnFracOut,
+                   Float_t ElecEta,
+                   Float_t ElecPhi,
+                   Float_t ElecEtotOverPin,
+                   Float_t ElecChi2NormGSF,
+                   Float_t ElecChi2NormKF,
+                   Float_t ElecGSFNumHits,
+                   Float_t ElecKFNumHits,
+                   Float_t ElecGSFTrackResol,
+                   Float_t ElecGSFTracklnPt,
+                   Float_t ElecPin,
+                   Float_t ElecPout,
+                   Float_t ElecEecal,
+                   Float_t ElecDeltaEta,
+                   Float_t ElecDeltaPhi,
+                   Float_t ElecMvaInSigmaEtaEta,
+                   Float_t ElecMvaInHadEnergy,
+                   Float_t ElecMvaInDeltaEta
+                  );
+
+   // CV: this function can be called for all categories
+   double MVAValue(const reco::PFTau& thePFTau, 
+		   const reco::GsfElectron& theGsfEle);
+   // CV: this function can be called for category 1 only !!
+   double MVAValue(const reco::PFTau& thePFTau);
+   
+ private:   
+
+   double dCrackEta(double eta);
+   double minimum(double a,double b);
+   double dCrackPhi(double phi, double eta);
+
+   bool isInitialized_;
+   bool loadMVAfromDB_;
+   edm::FileInPath inputFileName_;
+   
+   std::string mvaName_NoEleMatch_woGwoGSF_BL_;
+   std::string mvaName_NoEleMatch_wGwoGSF_BL_;
+   std::string mvaName_woGwGSF_BL_;
+   std::string mvaName_wGwGSF_BL_;
+   std::string mvaName_NoEleMatch_woGwoGSF_EC_;
+   std::string mvaName_NoEleMatch_wGwoGSF_EC_;
+   std::string mvaName_woGwGSF_EC_;
+   std::string mvaName_wGwGSF_EC_;
+   
+   Float_t* Var_NoEleMatch_woGwoGSF_Barrel_;
+   Float_t* Var_NoEleMatch_wGwoGSF_Barrel_;
+   Float_t* Var_woGwGSF_Barrel_;
+   Float_t* Var_wGwGSF_Barrel_;
+   Float_t* Var_NoEleMatch_woGwoGSF_Endcap_;
+   Float_t* Var_NoEleMatch_wGwoGSF_Endcap_;
+   Float_t* Var_woGwGSF_Endcap_;
+   Float_t* Var_wGwGSF_Endcap_;
+   
+   const GBRForest* mva_NoEleMatch_woGwoGSF_BL_;
+   const GBRForest* mva_NoEleMatch_wGwoGSF_BL_;
+   const GBRForest* mva_woGwGSF_BL_;
+   const GBRForest* mva_wGwGSF_BL_;
+   const GBRForest* mva_NoEleMatch_woGwoGSF_EC_;
+   const GBRForest* mva_NoEleMatch_wGwoGSF_EC_;
+   const GBRForest* mva_woGwGSF_EC_;
+   const GBRForest* mva_wGwGSF_EC_;
```

Comment:
Please don't use bare pointers like Float_t* and GBRForest*. I think the Var_ members are intended to be arrays, so you can use std::vector, while the mva_ members can be unique_ptr.